R/mlflow/R/remote.R

```diff
@@ -7,10 +7,10 @@
 #'
 #' @export
 mlflow_set_tracking_uri <- function(uri) {
-  if (!identical(.globals, uri)) {
-    .globals$tracking_uri <- uri
-    mlflow_set_active_experiment(NULL)
```

Comment:
When we reset the active experiment we should also reset the active run, otherwise another method will assume the run is active on another tracking server. These mechanisms should be baked into the setter methods. Tracking URI should also be coupled with active connection but I don't think I got around to that.

Reply:
This is working just fine since setting the connection to NULL causes downstream to re create the run... but yeah, being explicit here to reset the run makes sense.